spec/02-integration/03-db/12-dao_hooks_spec.lua

```diff
@@ -0,0 +1,280 @@
+local helpers = require "spec.helpers"
+local hooks = require "kong.hooks"
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("kong.db hooks [#" .. strategy .. "]", function()
+    local db, bp, s1, r1
+
+    lazy_setup(function()
+      bp, db = helpers.get_db_utils(strategy, {
+        "routes",
+        "services",
+      })
+
+      s1 = assert(bp.services:insert {
+        name = "s1",
+        url = "http://example.test",
+      })
+
+      r1 = assert(bp.routes:insert {
+        protocols = { "http" },
+        hosts = { "host1" },
+        service = s1,
+        name = "r1",
+      })
+    end)
+
+    describe("page_for", function()
+      local pre_hook = spy.new(function() end)
+      local post_hook = spy.new(function() end)
```

Comment:
Spies should be able to return true directly, you shouldn't have to "wrap them up":

```suggestion
      local pre_hook = spy.new(function() return true end)
      local post_hook = spy.new(function() return true end)
```

Reply:
This does not work. The hook must be a function, while `spy.new` returns a table with a __call metamethod I think.